spec.html

```diff
@@ -32151,12 +32162,19 @@ <h1>Array.prototype.join ( _separator_ )</h1>
           1. Else, let _sep_ be ? ToString(_separator_).
           1. Let _R_ be the empty String.
           1. Let _k_ be 0.
+          1. Let _realm_ be the current Realm Record.
+          1. Let _seen_ be _realm_.[[Seen]].
           1. Repeat, while _k_ &lt; _len_
             1. If _k_ &gt; 0, set _R_ to the string-concatenation of _R_ and _sep_.
             1. Let _element_ be ? Get(_O_, ! ToString(_k_)).
-            1. If _element_ is *undefined* or *null*, let _next_ be the empty String; otherwise, let _next_ be ? ToString(_element_).
+            1. If _seen_ contains _element_, then
+              1. Let _next_ be the empty String.
+              1. Append _element_ to _seen_.
```

Comment:
Why append `_element_` to `_seen_` if `_seen_` already contains it?
Should this step be under the `Else`?

Reply:
oh oops, yes :-) thanks, will fix.